content/05.main-text.md

```diff
@@ -412,7 +412,33 @@ we apply the following criteria to calculate _p~empiric~_ from summary statistic
 
 ### Rest API & backend
 
-<https://search-api.het.io/v1/>
+We created a backend application using Python's Django web framework.
+The source code is available in the [`connectivity-search-backend`](https://github.com/greenelab/connectivity-search-backend) repository.
+The primary role of the backend is manage a relational database and provide an API for requesting data.
+
+We define the database schema [using](https://github.com/greenelab/connectivity-search-backend/blob/af12f8cf2ad47d9a25ce8d1b7889390654eb3bb9/dj_hetmech_app/models.py "models.py for the dj_hetmech_app") Django's object-relational mapping framework (Figure @fig:database).
+We [import](https://github.com/greenelab/connectivity-search-backend/blob/af12f8cf2ad47d9a25ce8d1b7889390654eb3bb9/dj_hetmech_app/management/commands/populate_database.py "dj_hetmech_app populate_database management command") the data into a PostgreSQL database.
+Populating the database for all 2,205 metapaths up to length 3 was a prolonged operation, [taking](https://github.com/greenelab/connectivity-search-backend/pull/41#issuecomment-488054789) over 3 days.
+The majority of the time is spent populating the `DegreeGroupedPermutation` (37,905,389 rows) and `PathCount` (174,986,768 rows) tables.
+
+![
+**Schema for the connectivity search backend relational database.**
+<!--
+Consider touching this graph_models image up. Usage at
+https://github.com/django-extensions/django-extensions/blob/a7e86ec0a8708b5e0555f4a52f9be250a55c2012/django_extensions/management/commands/graph_models.py#L47
+-->
+](https://github.com/greenelab/connectivity-search-backend/raw/a91b25a0342050089c32ad2673e6a9c2f7ecad3f/media/models-schema.png){#fig:database width="5in"}
+
+We host a public API instance at <https://search-api.het.io>.
+Version 1 of the API exposes several endpoints that are used by the connectivity search frontend including queries for
+node details (`/v1/node`),
+node lookup (`/v1/nodes`),
+metapath information (`/v1/metapaths`),
+and path information (`/v1/paths`).
+The endpoints return JSON payloads.
+Producing results for these quieries relies on internal calls to both the PostgreSQL relational database as well as the pre-existing Hetionet v1.0 Neo4j graph database.
```

Comment:
This line has a few issues:
(1) typo: `queries`
(2) According to my English teacher in high school, either `both` should be deleted, or `as well as` should be changed to `and`.

Reply:
@dongbohu : I agree with your English teacher in high school! 👏